spec/event_tracer_spec.rb

```diff
@@ -98,8 +85,37 @@
         end
       end
 
-      context 'Logger fails gracefully when exception occurs' do
-        it_behaves_like 'error_in_logger_service_fails_gracefully'
+      context 'Logger fails when error occurs' do
+        before do
+          allow(mock_logger).to receive(selected_log_method).and_raise(RuntimeError.new('Runtime error in base logger'))
```

Comment:
Could we use `expect` instead of `allow`?

Reply:
Okay! For my understanding, when do we use `expect` vs `allow`? Or always prefer `expect`? Previously my practice was to use `expect` if I am testing a part of the flow while I would use `allow` if I'm just stubbing out a method.

Reply:
We prefer using `expect` so that error is raised if that method is not called.

Reply:
Got it! So always prefer `expect`.